readthedocs/builds/utils.py

```diff
@@ -61,3 +64,34 @@ def memcache_lock(lock_id, oid):
             # to lessen the chance of releasing an expired lock
             # owned by someone else.
             cache.delete(lock_id)
+
+
+def match_regex(pattern, text, timeout=1):
+    """
+    Find a match using regex.search.
+
+    .. note::
+
+        We use the regex module with the timeout
+        arg to avoid ReDoS.
+
+        We could use a finite state machine type of regex too,
+        but there isn't a stable library at the time of writting this code.
+    """
+    try:
+        match = regex.search(
+            pattern,
+            text,
+            # Compatible with the re module
+            flags=regex.VERSION0,
+            timeout=timeout,
+        )
+        return match
+    except TimeoutError:
+        log.warning(
+            'Timeout while parsing regex. pattern=%s, input=%s',
+            pattern, text,
+        )
+    except Exception as e:
+        log.info('Error parsing regex: %s', e)
```

Comment:
Similarly here -- seems bad to just log this exception. We probably want `log.exception`